core/cog-gamepad-manette.c

```diff
@@ -0,0 +1,325 @@
+        provider->connected = 0;
+    }
+
+    if (provider->disconnected > 0) {
+        g_signal_handler_disconnect(provider->monitor, provider->disconnected);
+        provider->disconnected = 0;
+    }
+}
+
+static int
+toStandardGamepadAxis(uint16_t axis)
+{
+    switch (axis) {
+    case ABS_X:
+        return WPE_GAMEPAD_AXIS_LEFT_STICK_X;
+    case ABS_Y:
+        return WPE_GAMEPAD_AXIS_LEFT_STICK_Y;
+    case ABS_RX:
+        return WPE_GAMEPAD_AXIS_RIGHT_STICK_X;
+    case ABS_RY:
+        return WPE_GAMEPAD_AXIS_RIGHT_STICK_Y;
+    default:
+        return -1;
+    }
+}
+
+static void
+on_absolute_axis(ManetteDevice *device, ManetteEvent *event, Gamepad *gamepad)
+{
+    uint16_t axis;
+    double   value;
+    int      std_axis;
+
+    if (!manette_event_get_absolute(event, &axis, &value))
+        return;
+
+    std_axis = toStandardGamepadAxis(axis);
+    if (std_axis != -1)
+        wpe_gamepad_dispatch_axis_changed(gamepad->server, (enum wpe_gamepad_axis) std_axis, value);
+}
+
+static int
+toStandardGamepadButton(uint16_t manetteButton)
+{
+    switch (manetteButton) {
+    case BTN_SOUTH:
+        return WPE_GAMEPAD_BUTTON_BOTTOM;
+    case BTN_EAST:
+        return WPE_GAMEPAD_BUTTON_RIGHT;
+    case BTN_NORTH:
+        return WPE_GAMEPAD_BUTTON_TOP;
+    case BTN_WEST:
+        return WPE_GAMEPAD_BUTTON_LEFT;
+    case BTN_TL:
+        return WPE_GAMEPAD_BUTTON_LEFT_SHOULDER;
+    case BTN_TR:
+        return WPE_GAMEPAD_BUTTON_RIGHT_SHOULDER;
+    case BTN_TL2:
+        return WPE_GAMEPAD_BUTTON_LEFT_TRIGGER;
+    case BTN_TR2:
+        return WPE_GAMEPAD_BUTTON_RIGHT_TRIGGER;
+    case BTN_SELECT:
+        return WPE_GAMEPAD_BUTTON_SELECT;
+    case BTN_START:
+        return WPE_GAMEPAD_BUTTON_START;
+    case BTN_THUMBL:
+        return WPE_GAMEPAD_BUTTON_LEFT_STICK;
+    case BTN_THUMBR:
+        return WPE_GAMEPAD_BUTTON_RIGHT_STICK;
+    case BTN_DPAD_UP:
+        return WPE_GAMEPAD_BUTTON_D_PAD_TOP;
+    case BTN_DPAD_DOWN:
+        return WPE_GAMEPAD_BUTTON_D_PAD_BOTTOM;
+    case BTN_DPAD_LEFT:
+        return WPE_GAMEPAD_BUTTON_D_PAD_LEFT;
+    case BTN_DPAD_RIGHT:
+        return WPE_GAMEPAD_BUTTON_D_PAD_RIGHT;
+    case BTN_MODE:
+        return WPE_GAMEPAD_BUTTON_CENTER;
+    default:
+        return -1;
+    }
+}
+
+static void
+on_button_press(ManetteDevice *device, ManetteEvent *event, Gamepad *gamepad)
+{
+    uint16_t button;
+    int      std_button;
+
+    if (!manette_event_get_button(event, &button))
+        return;
+
+    std_button = toStandardGamepadButton(button);
+    if (std_button != -1)
+        wpe_gamepad_dispatch_button_changed(gamepad->server, (enum wpe_gamepad_button) std_button, true);
+}
+
+static void
+on_button_release(ManetteDevice *device, ManetteEvent *event, Gamepad *gamepad)
+{
+    uint16_t button;
+    if (!manette_event_get_button(event, &button))
+        return;
+
+    wpe_gamepad_dispatch_button_changed(gamepad->server, toStandardGamepadButton(button), false);
```

Comment:
This is missing to check whether the button can be mapped to a `enum wpe_gamepad_button`, it should use the same logic as above.